mono/utils/mono-proclib.c

```diff
@@ -1053,38 +1052,38 @@ mono_pe_file_map (const gunichar2 *filename, gint32 *map_size, void **handle)
 			goto exit;
 		}
 
-		fd = open (located_filename, O_RDONLY, 0);
-		if (fd == -1) {
-			mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER_PROCESS, "%s: Error opening file %s (3): %s", __func__, filename_ext, strerror (errno));
+		if ((filed = mono_file_map_open (located_filename)) == NULL) {
+			mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER_PROCESS, "%s: Error opening file %s (3): %s", __func__, located_filename, strerror (errno));
 			goto exit;
 		}
 	}
-	else if (fd == -1) {
+	else if (filed == NULL) {
 		mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_IO_LAYER_PROCESS, "%s: Error opening file %s (3): %s", __func__, filename_ext, strerror (errno));
 		goto exit;
 	}
 
-	if (fstat (fd, &statbuf) == -1) {
+	fsize = mono_file_map_size (filed);
```

Comment:
Would it be worth putting a comment here that I am knowingly truncating a guint64 to a gint32?

Reply:
Maybe put assert that value fits in 32-bits?

Reply:
Consider making `fsize` and `map_size` unsigned or assert if the value is < 0;